openssl-dynamic/src/main/c/tcn_lock_rw.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#include "tcn_lock_rw.h"
+#include <shared_mutex>
+
+tcn_lock_rw_t tcn_lock_rw_new() {
+    // Once we switch to c++17 we should use std::shared_mutex
+    return (tcn_lock_rw_t) new std::shared_timed_mutex();
```

Comment:
IIRC in C++ it is ambiguous if you try to use `()` for the no argument constructor without the `new` keyword ... `new` may make the situation unambiguous but consider dropping the `()` which may be more `C++`ish.

http://stackoverflow.com/a/2318731